File: application/src/main/java/whenyourcar/application/service/car/CarSaleService.java
package whenyourcar.application.service.car;

import whenyourcar.domain.entity.CarSale;
import whenyourcar.domain.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import whenyourcar.application.dto.car.sale.CarSaleRequest;
import whenyourcar.application.dto.car.search.CarCommonResponse;

import java.util.Date;
import java.util.List;

public interface CarSaleService {
    public void postSaleCar(CarSaleRequest.CarSaleRequestDto carSaleRequest, User user, List<String> imagesURLs);
    public void patchCarToSaleCompleted(Long carId, String email);
    Page<CarCommonResponse.SearchResponseDto> searchCarsService(Pageable pageable, Date minAge, Date maxAge, Integer minMileage, Integer maxMileage, Integer minPrice, Integer maxPrice, String color);
    CarCommonResponse.SearchDescriptionSaleResponseDto searchDescriptionService(CarSale carSale);
    Page<CarCommonResponse.SearchDetailResponseDto> searchDetailCarsService(Pageable pageable, String manu, String model, String submodel, String grade);
    void increaseSaleCarView(Long carId);
    CarSale findCarSaleById(Long carId);
}

File: application/src/main/kotlin/howmuchcar/application/service/car/CarSaleService.kt
package howmuchcar.application.service.car

import howmuchcar.application.converter.car.CarSaleSearchDescResponse
import howmuchcar.application.converter.car.CarSearchDescResponse
import howmuchcar.application.converter.car.CarSearchDetailResponse
import howmuchcar.application.converter.car.CarSearchResponse
import howmuchcar.application.dto.car.CarSaleRequest
import howmuchcar.domain.entity.Car
import howmuchcar.domain.entity.CarSale
import howmuchcar.domain.entity.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.LocalDate
import java.util.*

interface CarSaleService {
    fun postSaleCar(carSaleRequest: CarSaleRequest,
                    user: User,
                    imageURL: List<String>)

    fun patchCarToSaleCompleted(carId: Long, user: User)

    fun searchCarSale(pageable: Pageable,
                      minAge: LocalDate?,
                      maxAge: LocalDate?,
                      minMileage: Int?,
                      maxMileage: Int?,
                      minPrice: Int?,
                      maxPrice: Int?,
                      color: String?): Page<CarSearchResponse>

    fun searchCarSaleDescription(carSale: CarSale): CarSaleSearchDescResponse

    fun searchCarSaleDetail(pageable: Pageable,
                            manu: String?,
                            model: String?,
                            submodel: String?,
                            grade: String?): Page<CarSearchDetailResponse>

    fun findCarSaleById(
        carId: Long,
    ): CarSale

}